oauth2client/contrib/appengine.py

```diff
@@ -476,18 +468,14 @@ def _parse_state_value(state, user):
         state: string, The value of the state parameter.
         user: google.appengine.api.users.User, The current user.
 
-    Raises:
-        InvalidXsrfTokenError: if the XSRF token is invalid.
-
     Returns:
-        The redirect URI.
+        The redirect URI, or None if XSRF token is not valid.
     """
     uri, token = state.rsplit(':', 1)
-    if not xsrfutil.validate_token(xsrf_secret_key(), token, user.user_id(),
-                                   action_id=uri):
-        raise InvalidXsrfTokenError()
-
-    return uri
+    if xsrfutil.validate_token(xsrf_secret_key(), token, user.user_id(),
+                               action_id=uri):
+        return uri
+    return
```

Comment:
Make it explicitly explicit by including the `None` in `return None`. Also put it in an `else:` to visually mark its exclusivity with the other branch of the `if`.